packages/react-experiments/src/components/UnifiedPicker/UnifiedPicker.tsx

```diff
@@ -267,164 +279,226 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
-            input.current.clear();
+    },
+    [focusedItemIndices.length, selectedItemsListGetItemCopyText, selection],
+  );
+
+  const _onInputFocus = React.useCallback(
+    (ev: React.FocusEvent<HTMLInputElement | Autofill>): void => {
+      unselectAll();
+      inputPropsOnFocus?.(ev as React.FocusEvent<HTMLInputElement>);
+    },
+    [inputPropsOnFocus, unselectAll],
+  );
+
+  const _onInputClick = React.useCallback(
+    (ev: React.MouseEvent<HTMLInputElement | Autofill>) => {
+      unselectAll();
+      showPicker(true);
+      inputPropsOnClick?.(ev as React.MouseEvent<HTMLInputElement>);
+    },
+    [inputPropsOnClick, showPicker, unselectAll],
+  );
+
+  const _onInputChange = React.useCallback(
+    (value: string, composing?: boolean, resultItemsList?: T[]) => {
+      if (!composing) {
+        // update query string
+        setQueryString(value);
+        // If we now have no query string, we want to deselect any selected item in the picker
+        if (value === '') {
+          clearPickerSelectedIndex();
+        }
+        // if nothing is selcted and the user has typed, selected the first picker item
+        else if (focusItemIndex === -1 && headerItemIndex === -1 && footerItemIndex === -1) {
+          setFocusItemIndex(0);
+        }
+        !isSuggestionsShown ? showPicker(true) : null;
+        if (!resultItemsList) {
+          resultItemsList = [];
+        }
+        if (onInputChange) {
+          onInputChange(value, composing, resultItemsList);
+          clearQueryString();
+          if (resultItemsList && resultItemsList.length > 0) {
+            addItems(resultItemsList);
+            showPicker(false);
+            // Clear the input
+            if (input.current) {
+              input.current.clear();
+            }
           }
         }
       }
-    }
-  };
-  const _onPaste = (ev: React.ClipboardEvent<Autofill | HTMLInputElement>) => {
-    if (props.onPaste) {
-      const inputText = ev.clipboardData.getData('Text');
-      ev.preventDefault();
-      // Pass current selected items
-      props.onPaste(inputText, selectedItems);
-      setSelectedItems(selectedItems);
-      selection.setItems(selectedItems);
-    }
-  };
+    },
+    [
+      addItems,
+      clearQueryString,
+      isSuggestionsShown,
+      onInputChange,
+      setQueryString,
+      showPicker,
+      clearPickerSelectedIndex,
+      setFocusItemIndex,
+      focusItemIndex,
+      footerItemIndex,
+      headerItemIndex,
+    ],
+  );
+
+  const _onPaste = React.useCallback(
+    (ev: React.ClipboardEvent<Autofill | HTMLInputElement>) => {
+      if (onPaste) {
+        const inputText = ev.clipboardData.getData('Text');
+        ev.preventDefault();
+        // Pass current selected items
+        onPaste(inputText, selectedItems);
+        setSelectedItems(selectedItems);
+        selection.setItems(selectedItems);
+      }
+    },
+    [onPaste, selectedItems, selection, setSelectedItems],
```

Comment:
why add setSelectedItems to dependencies? State setters are supposed to have stable references, I believe.

Reply:
unclear honestly. this version of lint really wants absolutely everything in the dependencies which is why it's there

Reply:
Oh it doesn't come from a UseState setter but from from a custom hook, nvm.